explorer/common/arena.h

```diff
@@ -61,35 +119,162 @@ class Arena {
 
   // Templated destruction of a pointer.
   template <typename T>
-  class ArenaEntryTyped : public ArenaEntry {
-   public:
-    struct WriteAddressHelper {};
+  class ArenaEntryTyped;
```

Comment:
Given the reference to llvm::Any for TypeId, I was looking and now wondering -- can this be replaced with llvm::Any?